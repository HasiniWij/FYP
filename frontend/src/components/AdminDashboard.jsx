import React, { useState,useEffect  }  from 'react';
import { useHistory } from "react-router-dom";
import './Style.css';
import logo from '../resources/logo.png'; 

export default function AdminDashboard() {
  const history = useHistory();
  const [authorized,setAuthorized]=useState(false)
  const homePage = () => {
    history.push("/adminDashboard")
  }
  const studentsPage = () => {
    history.push("/AdminStudentList")
  }
  const supervisorPage = () => {
    history.push("/adminSupervisorList")
  }
  const matchSupervisorsStudentsPage = () => {
    history.push("/matchSupervisorsStudents")
  }

  useEffect(() => {
    const role = localStorage.getItem('role');
    if(role=='admin') setAuthorized(true)
  },[]);

  return (
    <div>
    {authorized?
    <div class="container">
      <div class="row">
        <div class="col-2">
        <button class='logoButton' onClick={homePage}> <img src={logo} alt='logo'/>  </button>
        </div>
        <div class="d-flex justify-content-center title col-8">
          <h1> Admin Dashboard</h1>
        </div>
      </div>
      <div class='row dashboardItemRow largeMarginTop'>
        <div class='offset-3 col-3'>
          <button class='dashboardItem orangeBackground' onClick={supervisorPage}> Supervisor List</button>
        </div>
        <div class='col-3'>
          <button class='dashboardItem pinkBackground' onClick={studentsPage}> Student List</button>
        </div>
      </div>
      <div class='row dashboardItemRow'>
        <div class='offset-3 col-3'>
          <button class='dashboardItem blueBackground'  onClick={matchSupervisorsStudentsPage}>Match supervisors with students</button>
        </div>
      </div>
    </div>
     : 
     <h1 className='unauthorized'>
       401 authorization required
     </h1>
     }
     </div>
  );
}