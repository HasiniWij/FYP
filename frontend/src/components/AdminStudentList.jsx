import React, { useState, useEffect } from 'react';
import { useHistory } from "react-router-dom";
import useLoader from "../hooks/useLoader";
import './Style.css';
import logo from '../resources/logo.png';
import search from '../resources/search.png';
import axios from 'axios';

export default function AdminStudentList() {

  const [students, setStudents] = useState([]);
  const [allStudents, setAllStudents] = useState([]);
  const [loader, showLoader, hideLoader] = useLoader();
  const [authorized, setAuthorized] = useState(false);
  const [searchWord, setSearchWord] = useState('')
  const history = useHistory();
  const adminDashboard = () => {
    history.push("/adminDashboard")
  }

  useEffect(() => {
    const role = localStorage.getItem('role');
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    const token = localStorage.getItem('userToken');
    if (role == 'admin' && isLoggedIn === 'true') setAuthorized(true)
    showLoader();
    axios.get(`http://127.0.0.1:8000/api/students`, { headers: { "Authorization": `Bearer ${token}` } })
      .then(res => {
        hideLoader();
        setStudents(res.data.students)
        setAllStudents(res.data.students)
      })
      .catch((error) => {
        if (error.response.status == 401) {
          setAuthorized(false);
          localStorage.setItem('isLoggedIn', false);
        }
      });
  }, []);

  const studentCards = students.map((student, index) =>
    <div className='offset-1 col-3 supervisorCard' key={index}>
      <h5>{student.name}</h5>
      <div className='row'>
        <div className="col-7">
          <h5>{student.universityId} </h5>
        </div>
        {/* Luxury feature */}
        {/* <div className="col-5">
        <button className='adminIcon'> <img src={bin}/>  </button>
        <button className='adminIcon'> <img src={disable}/>  </button>
        </div> */}
      </div>
    </div>
  );

  const searchWordChange = event => {
    setSearchWord(event.target.value.trim());
  }
  const onSearch = () => {
    const filteredStudents = allStudents.filter(student => student.universityId.includes(searchWord));
    setStudents(filteredStudents);
  }
  return (
    <div>
      {authorized ?
        <div className="container">
          <div className="row marginTop">
            <div className="col-3">
              <button className='logoButton' onClick={adminDashboard}> <img src={logo} />  </button>
            </div>
            <div className="d-flex justify-content-center title col-5">
              <h1> Students List</h1>
            </div>
            <div className='col-4 searchBar'>
              <div className="input-group">
                <input type="search" className="form-control" onChange={searchWordChange} value={searchWord} placeholder="Search for student by ID" />
                <div className="input-group-append">
                  <button className="btn btn-outline-secondary" onClick={onSearch} type="button">
                    <img src={search} />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div className='row'>
            {studentCards}
          </div>
          {loader}
          <div className='row largeMarginTop marginBottom'>
            <div className='offset-5 col-2'>
              <button className='secondaryButton' onClick={adminDashboard}>Back to dashboard</button>
            </div>
          </div>
        </div>
        :
        <h1 className='unauthorized'>
          401 authorization required
        </h1>
      }
    </div>
  );
}