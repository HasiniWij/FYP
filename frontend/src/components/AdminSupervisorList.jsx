import React, { useState, useEffect } from 'react';
import { useHistory } from "react-router-dom";
import useLoader from "../hooks/useLoader";
import './Style.css';
import logo from '../resources/logo.png';
import bin from '../resources/bin.png';
import disable from '../resources/disable.png';
import search from '../resources/search.png';
import edit from '../resources/edit.png';
import axios from 'axios';

export default function AdminSupervisorList() {

  const [supervisors, setSupervisors] = useState([]);
  const [newCapacity, setNewCapacity] = useState('');
  const [editSupervisorId, setEditSupervisorId] = useState('');
  const [authorized, setAuthorized] = useState(false)
  const history = useHistory();
  const [loader, showLoader, hideLoader] = useLoader();
  const adminDashboard = () => {
    history.push("/adminDashboard")
  }
  useEffect(() => {
    const role = localStorage.getItem('role');
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    const token = localStorage.getItem('userToken');
    if(role=='admin' && isLoggedIn) setAuthorized(true)
    showLoader();
    axios.get(`http://127.0.0.1:8000/api/supervisors`, { headers: { "Authorization": `Bearer ${token}` } })
      .then(res => {
        setSupervisors(res.data)
        hideLoader();
      })
      .catch((error) => {
        if (error.response.status == 401) {
          hideLoader();
          setAuthorized(false);
          localStorage.setItem('isLoggedIn', false);
        }
      })
  }, []);
  const editCapacity = (supervisor) => {
    setEditSupervisorId(supervisor.id);
    setNewCapacity(supervisor.capacity)
  }
  const onCapacityChange = event => {
    setNewCapacity(event.target.value);
  }
  const saveCapacity = (supervisor) => {
    const token = localStorage.getItem('userToken');
    axios.post(`http://127.0.0.1:8000/api/updateCapacity`, {
      'id': supervisor.id,
      'capacity': newCapacity,
    }, { headers: { "Authorization": `Bearer ${token}` } })
      .then(res => {
        for (const supervisor of supervisors) {
          if (supervisor.id == editSupervisorId) {
            supervisor.capacity = newCapacity;
            setEditSupervisorId("");
            return;
          }
        }
      })
  }
  const cancelEdit = () => {
    setEditSupervisorId("");
  }

  const validateInput = (evt) => {
    (evt.key === 'e' || evt.key === '-' || evt.key === '.') && evt.preventDefault()
  }

  const supervisorCard = supervisors.map((supervisor) =>
    <div class='offset-1 col-5 supervisorCard'>
      <h5>{supervisor.name}</h5>
      <h5>{supervisor.email}</h5>
      <div class='row'>
        <div class="col-6">
          <h5>Number of students: {editSupervisorId !== supervisor.id && supervisor.capacity} </h5>
        </div>
        {editSupervisorId == supervisor.id ?
          <>
            <div class='col-2'>
              <input type="number" onKeyDown={(event) => validateInput(event)}
                class="form-control capacityInput" onChange={onCapacityChange} value={newCapacity} />
            </div>
            <div class='col-2'>
              <button class='primaryButton capacityInput'
                onClick={() => saveCapacity(supervisor)}>Save</button>
            </div>
            <div class='col-1 capacityInput'>
              <button class='secondaryButton' onClick={cancelEdit}>Cancel</button>
            </div>
          </>
          :
          <div class='col-2'>
            <button class='adminIcon editIcon' onClick={() => editCapacity(supervisor)}> <img src={edit} />  </button>
          </div>
        }
        {/* Luxury feature */}
        {/* <div class="col-3">
        <button class='adminIcon'> <img src={bin}/>  </button>
        <button class='adminIcon'> <img src={disable}/>  </button>
        </div> */}
      </div>
    </div>
  );

  return (
    <div>
      {authorized ?
        <div class="container">
          <div class="row marginTop">
            <div class="col-3">
              <button class='logoButton' onClick={adminDashboard}> <img src={logo} />  </button>
            </div>
            <div class="d-flex justify-content-center title col-5">
              <h1> Supervisors List</h1>
            </div>
            <div class='col-4 searchBar'>
              <div class="input-group">
                <input type="search" class="form-control" placeholder="Search for supervisor by ID" />
                <div class="input-group-append">
                  <button class="btn btn-outline-secondary" type="button">
                    <img src={search} />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div class='row marginTop'>
            {supervisorCard}
          </div>
          {loader}
          <div class='row largeMarginTop'>
            <div class='offset-5 col-2'>
              <button class='secondaryButton' onClick={adminDashboard}>Back to dashboard</button>
            </div>
          </div>
        </div>
        :
        <h1 className='unauthorized'>
          401 authorization required
        </h1>
      }
    </div>
  );
}