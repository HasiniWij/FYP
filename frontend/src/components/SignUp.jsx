import React, { useState } from 'react';
import logo from '../resources/logo.png';
import './Style.css';
import axios from 'axios';
import { Link } from "react-router-dom";
import { useHistory } from "react-router-dom";

export default function SignUp() {

  const history = useHistory();

  const [fullName, setFullName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');

  const onNameChange = event => {
    setFullName(event.target.value);
  };
  const onEmailChange = event => {
    setEmail(event.target.value);
  };
  const onPasswordChange = event => {
    setPassword(event.target.value);
  };

  const register = () => {
    const user = { 'name': fullName, 'email': email, 'password': password }
    axios.post(`http://127.0.0.1:8000/api/register`, user)
      .then(res => {
        localStorage.setItem('userToken', res.data.access_token)
        localStorage.setItem('userId', res.data.user.id)
        localStorage.setItem('role', res.data.user.role)
        localStorage.setItem('isLoggedIn', true)
        if (res.data.user.role === 'supervisor') {
          history.push("/supervisorDashboard")
        }
        else if (res.data.user.role === 'student') {
          history.push("/studentDashboard")
        }
        else if (res.data.user.role === 'admin') {
          history.push("/adminDashboard")
        }
      })
      .catch(error => {
        setEmail('');
        setPassword('');
        setFullName('');
        let message =''
        if(error.response.data.errors['email']){
          let index = 0;
          const elements = error.response.data.errors['email'];
          while (index <elements .length) 
          {
              message = message + ' ' +elements[index]
              index++;
          }
        }
        if(error.response.data.errors['name']){
          let index = 0;
          const elements = error.response.data.errors['name'];
          while (index <elements .length) 
          {
              message = message + ' ' +elements[index]
              index++;
          }
        }
        if(error.response.data.errors['password']){
          let index = 0;
          const elements = error.response.data.errors['password'];
          while (index <elements .length) 
          {
              message = message + ' ' +elements[index]
              index++;
          }
        }
        setError(message);
      });
  }

  return (
    <div className="container authFormContainer">
      <div className='row'>
        <div className='col-6 authFormLeft'>
          <img src={logo} className='authLogo' />
          <h1 className='authFormTitle'>Welcome to FYPHelper</h1>
        </div>
        <div className='col-4 offset-2 authFormRight'>
          <div className='row'>
            <h5>Full Name</h5>
            <input type="text" className="form-control" onChange={onNameChange} placeholder="Enter your full name" value={fullName} />
          </div>
          <div className='row largeMarginTop'>
            <h5>University Email</h5>
            <input type="email" className="form-control" onChange={onEmailChange} placeholder="Enter your email" value={email} />
          </div>
          <div className='row largeMarginTop'>
            <h5>Password</h5>
            <input type="password" className="form-control" onChange={onPasswordChange} placeholder="Enter your password" value={password} />
          </div>
          <div className='row marginTop authError'>
            <p>{error}</p>
          </div>
          <div className='row authButton'>
            <button className='primaryButton' onClick={register}>Create Account</button>
          </div>
          <div className='row largeMarginTop authNavigate'>
            <p>Already have an account?
              <Link className='authLink' to={'signIn'}>
                Log in
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}