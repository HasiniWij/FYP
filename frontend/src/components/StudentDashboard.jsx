import React, { useState, useEffect } from 'react';
import { useHistory } from "react-router-dom";
import './Style.css';
import logo from '../resources/logo.png';
import logoutIcon from '../resources/logout.png';
import profile from '../resources/profile.png';
import axios from 'axios';
import useLoader from "../hooks/useLoader";

export default function StudentDashboard() {
  const history = useHistory();
  const [loader, showLoader, hideLoader] = useLoader();
  const [authorized, setAuthorized] = useState(false);

  useEffect(() => {
    const role = localStorage.getItem('role');
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    if (role == 'student'  && isLoggedIn==='true') setAuthorized(true)
  }, []);

  const bookMeetingPage = () => {
    history.push("/studentMeeting")
  }
  const requestSupervisionPage = () => {
    history.push("/requestSupervision")
  }
  const supervisorListPage = () => {
    history.push("/supervisorsList")
  }
  const profilePage = () => {
    history.push("/profile")
  }
  const homePage = () => {
    history.push("/studentDashboard")
  }
  const proposedProjectsPage = () => {
    history.push("/proposedProjects")
  }
  const logsPage = () => {
    history.push("/logs")
  }

  const logout = () => {
    showLoader();
    const token = localStorage.getItem('userToken');
    axios.get(`http://127.0.0.1:8000/api/logout/`, { headers: { "Authorization": `Bearer ${token}` } })
      .then(res => {
        hideLoader();
        localStorage.clear();
        history.push('/signIn');
      })
  }

  return (
    <div>
      {authorized ?
        <div className="container">
          <div className="row">
            <div className="col-2">
              <button className='logoButton' onClick={homePage}> <img src={logo} />  </button>
            </div>
            <div className="d-flex justify-content-center title col-7">
              <h1> Student Dashboard</h1>
            </div>
            <div className="col-1 profileIconArea">
              <button className='profileButton' onClick={profilePage}><img className='profileIcon' src={profile} /></button>
            </div>
            <div className="col-1 profileIconArea">
              <button className='logoutIcon' onClick={logout} ><img src={logoutIcon} /></button>
            </div>
          </div>
          <div className='row dashboardItemRow largeMarginTop'>
            <div className='col-2'> </div>
            <div className='col-3'>
              <button className='dashboardItem purpleBackground' onClick={proposedProjectsPage}>View proposed project topics</button>
            </div>
            <div className='col-3'>
              <button className='dashboardItem pinkBackground' onClick={supervisorListPage}>View supervisor list</button>
            </div>
            <div className='col-3'>
              <button className='dashboardItem greenBackground' onClick={bookMeetingPage}>Book meeting time slots</button>

            </div>
          </div>
          <div className='row dashboardItemRow'>
            <div className='col-2'> </div>
            <div className='col-3'>
              <button className='dashboardItem blueBackground' onClick={requestSupervisionPage}>Request for a supervisor</button>
            </div>
            <div className='col-3'>
              <button className='dashboardItem yellowBackground' onClick={logsPage}>Meeting logs</button>
            </div>
            <div className='col-3'>
              <button className='dashboardItem orangeBackground'>Messages</button>
            </div>
          </div>
          {loader}
        </div> :
        <h1 className='unauthorized'>
          401 authorization required
        </h1>
      }
    </div>
  );
}