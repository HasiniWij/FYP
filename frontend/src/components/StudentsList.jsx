import React,{useState, useEffect } from 'react';
import { useHistory } from "react-router-dom";
import useLoader from "../hooks/useLoader";
import axios from 'axios';
import './Style.css';
import logo from '../resources/logo.png'; 

export default function StudentsList() {

  const history = useHistory();
  const [loader, showLoader, hideLoader] = useLoader(); 

  const [students, setStudents] = useState([]);
  const [authorized,setAuthorized]=useState(false);

  useEffect(() => { 
    const token = localStorage.getItem('userToken');
    const role = localStorage.getItem('role');
    const userId = localStorage.getItem('userId');
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    if (role == 'supervisor' && isLoggedIn) setAuthorized(true);
    showLoader();
    axios.get(`http://127.0.0.1:8000/api/supervisorStudents/`+userId,{ headers: {"Authorization" : `Bearer ${token}`}})
      .then(res => {
        hideLoader()
        setStudents(res.data);
      })
      .catch((error) => {
        if (error.response.status == 401) {
          hideLoader();
          setAuthorized(false);
          localStorage.setItem('isLoggedIn', false);
        }
      })
  },[]);
  
  const supervisorDashboard = () => {
    history.push("/supervisorDashboard")
  }
  const profilePage = () => {
    history.push("/profile")
  }
  const logsPage = (name) => {
    history.push("/logs/"+name);
  }
  
  const studentCards = students.map((student) =>
    <button class='offset-1 col-2 studentCard text-center' onClick={() => logsPage(student.name)}>
      <h6>{student.name}</h6>
    </button>
  );

  return (
    <div>
    {authorized?
    <div class="container">
      <div class="row marginTop">
        <div class="col-2">
        <button class='logoButton' onClick={supervisorDashboard}> <img src={logo}/>  </button>
        </div>
        <div class="d-flex justify-content-center title col-8">
          <h1> Students</h1>
        </div>
        <div class="col-2 profileIconArea">
          <button class='profileButton' onClick={profilePage}>H</button>
        </div>
      </div>
      <div class='row marginTop'>
        {studentCards}
      </div>
      {loader}
      <div class='row largeMarginTop'>
        <div class='offset-5 col-2'>
          <button class='secondaryButton' onClick={supervisorDashboard}>Back to dashboard</button>
        </div>
      </div>
    </div>: 
    <h1 className='unauthorized'>
      401 authorization required
    </h1>
    }
    </div>
  );
}