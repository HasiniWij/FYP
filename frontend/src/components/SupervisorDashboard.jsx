import React, { useState, useEffect } from 'react';
import { useHistory } from "react-router-dom";
import './Style.css';
import logo from '../resources/logo.png';
import axios from 'axios';
import logoutIcon from '../resources/logout.png';
import useLoader from "../hooks/useLoader";
import profile from '../resources/profile.png';

export default function SupervisorDashboard() {
  const [loader, showLoader, hideLoader] = useLoader();
  const history = useHistory();
  const [authorized, setAuthorized] = useState(false);

  useEffect(() => {
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    const role = localStorage.getItem('role');
    if (role == 'supervisor' && isLoggedIn === 'true') setAuthorized(true);
  }, []);

  const profilePage = () => {
    history.push("/profile")
  }
  const homePage = () => {
    history.push("/supervisorDashboard")
  }
  const studentsPage = () => {
    history.push("/studentsList")
  }
  const supervisionRequestsPage = () => {
    history.push("/supervisionRequests")
  }
  const scheduleMeetingsPage = () => {
    history.push("/scheduleMeetings")
  }
  const logout = () => {
    showLoader();
    const token = localStorage.getItem('userToken');
    axios.get(`http://127.0.0.1:8000/api/logout/`, { headers: { "Authorization": `Bearer ${token}` } })
      .then(res => {
        hideLoader();
        localStorage.clear();
        history.push('/signIn');
      })
  }

  return (
    <div>
      {authorized ?
        <div className="container">
          <div className="row">
            <div className="col-2">
              <button className='logoButton' onClick={homePage}> <img src={logo} alt='logo' />  </button>
            </div>
            <div className="d-flex justify-content-center title col-7">
              <h1> Supervisor Dashboard</h1>
            </div>
            <div className="col-1 profileIconArea">
              <button className='profileButton' onClick={profilePage}>
                <img className='profileIcon' src={profile} />
              </button>
            </div>
            <div className='col-1 profileIconArea'>
              <button className='logoutIcon' onClick={logout} ><img src={logoutIcon} /></button>
            </div>
          </div>
          <div className='row dashboardItemRow largeMarginTop'>
            <div className='offset-3 col-3'>
              <button className='dashboardItem orangeBackground' onClick={studentsPage}>Students</button>
            </div>
            <div className='col-3'>
              <button className='dashboardItem blueBackground' onClick={scheduleMeetingsPage}>Schedule meetings</button>
            </div>
          </div>
          <div className='row dashboardItemRow'>
            <div className='offset-3 col-3'>
              <button className='dashboardItem pinkBackground' onClick={supervisionRequestsPage}>Requests for supervision</button>
            </div>
            <div className='col-3'>
              <button className='dashboardItem greenBackground'>Messages</button>
            </div>
          </div>
          {loader}
        </div> :
        <h1 className='unauthorized'>
          401 authorization required
        </h1>
      }
    </div>
  );
}