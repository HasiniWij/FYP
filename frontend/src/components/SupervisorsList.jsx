import React,{useState, useEffect } from 'react';
import { useHistory } from "react-router-dom";
import axios from 'axios';
import './Style.css';
import logo from '../resources/logo.png'; 

export default function SupervisorsList() {

  const history = useHistory();
  const studentDashboard = () => {
    history.push("/studentDashboard")
  }
  const profilePage = () => {
    history.push("/profile")
  }
  const [supervisors, setSupervisors] = useState([]);
  
  useEffect(() => { 
    const token = localStorage.getItem('userToken');
    axios.get(`http://127.0.0.1:8000/api/supervisors`,{ headers: {"Authorization" : `Bearer ${token}`}})
      .then(res => {
        setSupervisors(res.data);
      })
  },[]);
  
  const supervisorCards = supervisors.map((supervisor) =>
    <div key={supervisor.name} className='offset-1 col-5 supervisorCard'>
      <h6>Name: {supervisor.name}</h6>
      <h6>Interested areas: {supervisor.interests} </h6>
      <button className='primaryButton projectIdeaButton'>Go to project ideas</button>
    </div>
  );

  return (
    <div className="container">
      <div className="row">
        <div className="col-2">
        <button className='logoButton' onClick={studentDashboard}> <img src={logo}/>  </button>
        </div>
        <div className="d-flex justify-content-center title col-8">
          <h1> Supervisors List</h1>
        </div>
        <div className="col-2 profileIconArea">
          <button className='profileButton' onClick={profilePage}>H</button>
        </div>
      </div>
      <div className='row'>
        {supervisorCards}
      </div>
      <div className='row marginTop'>
        <div className='offset-5 col-2'>
          <button className='secondaryButton' onClick={studentDashboard}>Back to dashboard</button>
        </div>
      </div>
    </div>
  );
}