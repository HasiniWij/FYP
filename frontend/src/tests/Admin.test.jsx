import React from "react";
import renderer from "react-test-renderer";
import axios from 'axios';
import AdminDashboard from '../components/AdminDashboard';
import StudentDashboard from "../components/StudentDashboard";
import SupervisorDashboard from "../components/SupervisorDashboard";
import SignUp from "../components/SignUp";
import AdminStudentList from "../components/AdminStudentList";
import StudentsList from "../components/StudentsList";
import SupervisorsList from "../components/SupervisorsList";
import AdminSupervisorList from "../components/AdminSupervisorList";

const students = [
  {
   'universityId':'w245678',
   'name':'Jon Doe'
  },
  {
    'universityId':'w24d678',
    'name':'Tecla jof'
   },
]
const supervisors =[
  // {
  //   'interests':'web development',
  //   'name':'Jon Doe'
  //  },
   {
     'interests':'w24d678',
     'name':'mobile development, data science',
     "email":'sss',
     "capacity":3,
     "id":'22',
    }
]
jest.mock("axios")
describe("Jest Snapshot testing suite", () => {
  it("Snapshot testing AdminDashboard", () => {
    const domTree = renderer.create(<AdminDashboard />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Snapshot testing StudentDashboard", () => {
    const domTree = renderer.create(<StudentDashboard />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Snapshot testing SupervisorDashboard", () => {
    const domTree = renderer.create(<SupervisorDashboard />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Snapshot testing AdminStudentList", async () => {
    axios.get.mockImplementationOnce(() => Promise.resolve({ data: students }))
    const tree = renderer.create(<AdminStudentList />);
    await Promise.resolve();
    expect(tree.toJSON()).toMatchSnapshot()
  })
  // it("Snapshot testing StudentsList", async () => {
  //   axios.get.mockImplementationOnce(() => Promise.resolve({ 'data': {'supervisors':supervisors}}))
  //   const tree = renderer.create(<AdminSupervisorList />);
  //   await Promise.resolve();
  //   expect(tree.toJSON()).toMatchSnapshot()
  // })
  // it("Matches DOM Snapshot", () => {
  //   const domTree = renderer.create(<SignUp />).toJSON();
  //   expect(domTree).toMatchSnapshot();
  // });
});
