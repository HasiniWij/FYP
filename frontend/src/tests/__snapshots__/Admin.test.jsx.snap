// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite Matches DOM Snapshot 2`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite Matches DOM Snapshot 3`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite Snapshot testing AdminDashboard 1`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite Snapshot testing AdminStudentList 1`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite Snapshot testing StudentDashboard 1`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite Snapshot testing SupervisorDashboard 1`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;

exports[`Jest Snapshot testing suite test 1`] = `
<div>
  <h1
    className="unauthorized"
  >
    401 authorization required
  </h1>
</div>
`;
